docs/usage/upload_files.rst

```diff
@@ -0,0 +1,199 @@
+..
+    This file is part of Invenio.
+    Copyright (C) 2015-2019 CERN.
+
+    Invenio is free software; you can redistribute it and/or modify it
+    under the terms of the MIT License; see LICENSE file for more details.
+
+
+Creating Buckets
+----------------
+
+In order to upload, modify or delete files, we first need to create a bucket.
+A bucket can be created by a POST request to the url prefix, i.e. /files.
+The response will contain the unique ID of the bucket.
+A bucket can have one or more tags which store extra metadata for that bucket.
+Each tag is uniquely identified by a key.
+
+First we create a bucket:
+
+.. code-block:: console
+
+   $ curl -X POST http://localhost:5000/files
+
+.. code-block:: json
+
+    {
+        "max_file_size": null,
+        "updated": "2019-05-16T13:07:21.595398+00:00",
+        "locked": false,
+        "links": {
+            "self": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e",
+            "uploads": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e?uploads",
+            "versions": "http://localhost:5000/files/cb8d0fa7-2349-484b-89cb-16573d57f09e?versions"
+        },
+        "created": "2019-05-16T13:07:21.595391+00:00",
+        "quota_size": null,
+        "id": "cb8d0fa7-2349-484b-89cb-16573d57f09e",
+        "size": 0
+    }
+
```

Comment:
So in general, IMHO, I would first explain the concepts. Imagine I don't know anything about Bucket/ObjectVersion. I need to have an overview of the data model first (we need a location, we need a bucket, then we can upload a file, that is represented by ObjectVersion/FileInstance)
WDYT?